scipy/sparse/__init__.py

```diff
@@ -322,3 +322,30 @@
 from scipy._lib._testutils import PytestTester
 test = PytestTester(__name__)
 del PytestTester
+
+# Register the sparsetools thread pool with threadpoolctl
+try:
+    import threadpoolctl
+
+    class _SparseToolsThreadPoolCtlController(threadpoolctl.LibController):
+        user_api = "scipy"
+        internal_api = "scipy_sparsetools"
+
+        filename_prefixes = ("_sparsetools",)
+
+        def get_num_threads(self):
+            return _sparsetools.get_workers()
```

Comment:
`_sparsetools` is undefined, looks like you're missing a `from . import _sparsetools` higher up.

Reply:
Done.